client/client_with_secret_test.go

```diff
@@ -215,3 +217,67 @@ func Test_GivenTokenSourceWithTokenThatWillNotExpireBetweenRequests_WhenTokenSto
 	mockedAuthService.AssertExpectations(t)
 	mockedClient.AssertExpectations(t)
 }
+
+func Test_GivenAServerThatRejectsHTTPCall_WhenAGetCallIsFired_ThenExpectTheClientToRenewAccessTokenOnce(t *testing.T) {
+	// Given
+	mockedAuthService := mocks.AuthService{}
+	mockedClient := mocks.Client{}
+
+	accessToken := "initial-access-token"
+
+	ts := startMockServer(t, &mockedAuthService, &mockedClient, accessToken, http.StatusOK, http.StatusUnauthorized)
+	defer ts.Close()
+
+	mockedAuthService.
+		On("Token").Return().
+		Twice()
+	mockedClient.
+		On("Test", "initial-access-token").Return().
+		Twice()
+
+	// When
+	c := client.NewWithSecret("my-client-id", "my-secret",
+		client.WithBaseURL(ts.URL)).HTTPClient()
+
+	// Then
+	resp, err := c.Get(ts.URL)
+	assert.NoError(t, err)
+	assert.Equal(t, http.StatusUnauthorized, resp.StatusCode)
+
+	mockedAuthService.AssertExpectations(t)
```

Comment:
We have expectation for `mockedAuthService` both as `.AssertExpectations(t)` and `.AssertNotCalled(t....` which is a bit hard to follow. Could we assert only one way? Ideally not using the `AssertExpectations`.